diff -up element-desktop-1.11.2/src/electron-main.ts.orig element-desktop-1.11.2/src/electron-main.ts
--- element-desktop-1.11.2/src/electron-main.ts	2022-08-03 17:10:33.955737022 +0800
+++ element-desktop-1.11.2/src/electron-main.ts	2022-08-03 17:12:04.790209296 +0800
@@ -204,6 +204,7 @@ async function setupGlobals(): Promise<v
     global.launcher = new AutoLaunch({
         name: global.vectorConfig.brand || 'Element',
         isHidden: true,
+        path: "/usr/bin/element-desktop",
         mac: {
             useLaunchAgent: true,
         },
